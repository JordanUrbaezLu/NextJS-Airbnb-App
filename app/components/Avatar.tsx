"use client";

import Image from "next/image";
import { useEffect } from "react";
import toast, { useToasterStore } from "react-hot-toast";

interface AvatarProps {
  src: string | null | undefined;
}

const Avatar: React.FC<AvatarProps> = ({ src }) => {
  // Add one useEffect for amount of max toasts
  const { toasts } = useToasterStore();

  const TOAST_LIMIT = 2;

  useEffect(() => {
    toasts
      .filter((t) => t.visible) // Only consider visible toasts
      .filter((_, i) => i >= TOAST_LIMIT) // Is toast index over limit?
      .forEach((t) => toast.dismiss(t.id)); // Dismiss â€“ Use toast.remove(t.id) for no exit animation
  }, [toasts]);
  // Add one useEffect for amount of max toasts

  return (
    <Image
      className="rounded-full"
      height="30"
      width="30"
      alt="Avatar"
      src={src || "/images/placeholder.jpg"}
    />
  );
};

export default Avatar;
