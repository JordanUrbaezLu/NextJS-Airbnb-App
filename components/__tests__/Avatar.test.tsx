import { render } from "@testing-library/react";
import Avatar from "../Avatar";

describe("Avatar", () => {
  test("Should render component correctly.", () => {
    const { container } = render(<Avatar src={undefined} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
