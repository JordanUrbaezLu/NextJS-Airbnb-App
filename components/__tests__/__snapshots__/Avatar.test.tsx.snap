// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Should render component correctly. 1`] = `
<img
  alt="Avatar"
  class="rounded-full"
  data-nimg="1"
  decoding="async"
  height="30"
  loading="lazy"
  src="/_next/image?url=%2Fimages%2Fplaceholder.jpg&w=64&q=75"
  srcset="/_next/image?url=%2Fimages%2Fplaceholder.jpg&w=32&q=75 1x, /_next/image?url=%2Fimages%2Fplaceholder.jpg&w=64&q=75 2x"
  style="color: transparent;"
  width="30"
/>
`;
